### DISCUSSION TITLE
Extending NavigationItem?

### DISCUSSION THREAD

**Discussion by KDederichs** [sulu/sulu#6629] [Q&A] [answered]
Hey,

I was wondering if it's possible to extend the NavigationItem with some extra properties.

I'm specifically looking to build a multi level collapsible menu that has different teasers at each level (like one or two of them with image/headline and a link to either external resources or a sulu resource)

Did anyone ever do something similar/knows what's the propper way to add extra data like that to the navigation? :)

**Comment by alexander-schranz** [comment, marked as answer]
The navigation has an additional flag called for loading also the `loadExcerpt`: https://docs.sulu.io/en/2.4/reference/twig-extensions/functions/sulu_navigation_root_tree.html

With that flag you will also get the excerpt image which you can use as a teaser image and so on.
The `depth` allows you to create the nested menu. To link to external resources create  a new page add it to the navigation context change the type to "external" and link to the external resource.

All other things are just frontend HTML/CSS/JS and there you are completely free and can create a menu the way you want it. Different website with different menus you can also see in the case studies: https://sulu.io/case-studies

**Reply by KDederichs** [reply]
Ahhhh that does make a lot of sense to use the excrept that way, thanks!
